README.md

```diff
@@ -65,5 +65,20 @@ Java unit testing is done with [JUnit 4](https://junit.org/junit4/).
 **To add another JS test**, simply add a JS file that ends with `.test.js` (ex: `script.test.js`) and it will automatically be run by Jest.
 **To add another Java test**, add a Java file to the testing directory `src/test/java/`. Be sure to follow the Java conventions for package directory structure, for example if you're testing a package in `com.google.opensesame.github`, the test should be placed in the directory `src/test/java/com/google/opensesame/github`.
 
+## User Authentication and Sign Up
+User signup consists of three parts:
+1. Website authentication (a user creates an account on our website)
+  * This is done using the [Users API](https://cloud.google.com/appengine/docs/standard/java/users)
+  * In the future this could be migrated to the [Firebase Admin SDK](https://firebase.google.com/docs/reference/admin) or similar OAuth-based authentication API such that GitHub authentication and website authentication could become a single step.
+2. GitHub authentication (a user verifies ownership of a GitHub account, which is then linked with the account on our website)
+  * This is done using [Firebase Authentication](https://firebase.google.com/docs/auth) with [GitHub OAuth](https://docs.github.com/en/developers/apps/about-apps#about-oauth-apps)
+  * An access token is granted to the user, which is then sent to the server during the profile creation step and verified on the backend. 
+  * Once the access token is verified, the user has proven ownership of that GitHub account, and it is then stored by ID in the website profile.
+3. Profile creation (a user with an authorized account can add more information about themselves)
```

Comment:
If, for now, "add more info about themselves" just means add interest tags, then I would probably rephrase this description.

Reply:
I definitely see what you're saying, but I wonder if that just presents an opportunity for the documentation to become out of date? As it stands now it still conveys enough information but is general enough to still apply with future changes.

Reply:
That's a good middleground, thanks! Will update.